youtube_dl/extractor/megatvcom.py

```diff
@@ -0,0 +1,186 @@
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        _is_article = video_id == 'None'
+        webpage = self._download_webpage(url,
+                                               'N/A' if _is_article else
+                                               video_id)
+        if _is_article:
+            video_id = self._match_article_id(webpage)
+        player_attrs = self._extract_player_attrs(webpage)
+        title = player_attrs.get('label') or self._og_search_title(webpage)
+        description = clean_html(get_element_by_class(
+            'article-wrapper' if _is_article else 'story_content',
+            webpage))
+        if not description:
+            description = self._og_search_description(webpage)
+        thumbnail = player_attrs.get('image') or \
+            self._og_search_thumbnail(webpage)
+        timestamp = unified_timestamp(self._html_search_meta(
+            'article:published_time', webpage))
+        try:
+            source = player_attrs['source']
+        except KeyError:
+            raise ExtractorError('no source found for %s' % video_id)
+        formats = self._extract_m3u8_formats(source, video_id, 'mp4') \
+            if determine_ext(source) == 'm3u8' else [source]
+        self._sort_formats(formats)
+        return {
+            'id': video_id,
+            'title': title,
+            'description': description,
+            'thumbnail': thumbnail,
+            'timestamp': timestamp,
+            'formats': formats,
+        }
+
+
+class MegaTVComEmbedIE(MegaTVComBaseIE):
+    IE_NAME = 'megatvcom:embed'
+    IE_DESC = 'megatv.com embedded videos'
+    _VALID_URL = r'https?://(?:www\.)?megatv.com/embed/?\?p=\d+'
+
+    _TESTS = [{
+        'url': 'https://www.megatv.com/embed/?p=2020520979',
+        'md5': '2ebe96661cb81854889053cebb661068',
+        'info_dict': {
+            'id': '520979',
+            'ext': 'mp4',
+            'title': 'md5:70eef71a9cd2c1ecff7ee428354dded2',
+            'description': 'md5:0209fa8d318128569c0d256a5c404db1',
+            'timestamp': 1634975747,
+            'upload_date': '20211023',
+        },
+    }, {
+        'url': 'https://www.megatv.com/embed/?p=2020534081',
+        'md5': 'f9a15e315acbf01b128e8efa3f75aab3',
+        'info_dict': {
+            'id': '534081',
+            'ext': 'mp4',
+            'title': 'md5:062e9d5976ef854d8bdc1f5724d9b2d0',
+            'description': 'md5:36dbe4c3762d2ede9513eea8d07f6d52',
+            'timestamp': 1636376351,
+            'upload_date': '20211108',
+        },
+    }]
+
+    @classmethod
+    def _extract_urls(cls, webpage, origin_url=None):
+        # make the scheme in _VALID_URL optional
+        _URL_RE = r'(?:https?:)?//' + cls._VALID_URL.split('://', 1)[1]
+        EMBED_RE = r'''(?x)
+            <iframe[^>]+?src=(?P<_q1>%(quot_re)s)
+                (?P<url>%(url_re)s)(?P=_q1)
+        ''' % {'quot_re': r'[\"\']', 'url_re': _URL_RE}
```

Comment:
No need to escape a double quote inside a single (in fact that might break this due to `r'`).

Reply:
Right. I am not sure about the breakage, again:

```py3
patterns = (r'["\']', r'[\"\']')
strings = ('"', '\"', "'", "\'")
ctx = 'iframe src=%shttps://whatever'
all(re.match(ctx % p, ctx % s) for p in patterns for s in strings) # True
```